File: problems/007/p007.go
package main

import "fmt"

func main() {

	var example_result int = example()

	fmt.Printf("Example=%d\n", example_result)

	var problem_result int = problem()

	fmt.Printf("Result=%d\n", problem_result)

}

func isPrime(num int) bool {

	if num < 4 {
		return true
	}

	for i := 2; i < num; i++ {

		if num%i == 0 {
			return false
		}

	}

	return true
}

func resolve(factor int) int {

	var last_prime int
	var prime_count int
	num := 2

	for prime_count < factor {

		is_prime := isPrime(num)

		if is_prime {
			last_prime = num
			prime_count++
		}

		num++

	}

	return last_prime
}

func example() int {
	return resolve(6)
}

func problem() int {
	return resolve(10001)
}

File: problems/007/p007.c
#include <stdio.h>
#include <math.h>

int isPrime(int n)
{
	if (n < 2)
	{
		return 0;
	}

	if (n == 2 || n == 3)
	{
		return 1;
	}

	if (n % 2 == 0 || n % 3 == 0)
	{
		return 0;
	}

	for (int i = 5; i <= sqrt(n); i += 6)
	{
		if (n % i == 0)
		{
			return 0;
		}
		else if (n % (i + 2) == 0)
		{
			return 0;
		}
	}

	return 1;
}

int resolve(int factor)
{
	int last_prime = 0;
	int prime_count = 0;
	int num = 2;

	while (prime_count < factor)
	{

		int is_prime = isPrime(num);

		if (is_prime)
		{
			last_prime = num;
			prime_count++;
		}

		num++;
	}

	return last_prime;
}

int main()
{

	int result = resolve(10001);

	printf("Result=%d\n", result);

	return 0;
}